src/app/timer-card/timer-card.component.ts

```diff
@@ -19,70 +10,76 @@ export type Status = {
   templateUrl: './timer-card.component.html',
   styleUrls: ['./timer-card.component.scss'],
 })
-export class TimerCardComponent implements AfterViewInit {
+export class TimerCardComponent implements AfterViewInit, OnDestroy {
   @Input() timeInSeconds!: number;
   @Input() id!: number;
   @Input() label!: string;
   color: string = 'primary';
-  isPaused: Boolean = true;
-  isCompleted: Boolean = false;
+  isPaused!: boolean;
+  isCompleted: boolean = false;
   countdown$!: Observable<number>;
   progress$!: Observable<number>;
-  private pauseClick$!: Observable<Status>;
-  private tick$ = timer(0, 1000);
+  private sub!: Subscription;
+
+  constructor(private ts: TimerService) {}
 
   ngAfterViewInit(): void {
-    this.pauseClick$ = fromEvent(
-      document.getElementById('' + this.id) as HTMLButtonElement,
-      'click'
-    ).pipe(
-      map(() => ({
-        active: !this.isPaused,
-      }))
-    );
-    this.countdown$ = this.timer(this.timeInSeconds);
-    this.progress$ = this.progressBarValue(this.timeInSeconds);
+    this.setStreams();
+  }
+
+  ngOnDestroy(): void {
+    this.sub.unsubscribe();
   }
 
   pauseTimer() {
     this.isPaused = !this.isPaused;
-    this.color = this.isPaused ? 'primary' : 'warn';
+    this.ts.updateTimerStatus(this.id, this.isPaused);
+    this.toggleButtonColor();
   }
 
   deleteTimer() {
-    //functionality will be added with the rxjs streams implementation.
+    this.ts.removeTimer(this.id);
   }
 
   reloadTimer() {
-    this.isCompleted = false;
     if (!this.isPaused) this.pauseTimer();
-    this.countdown$ = this.timer(this.timeInSeconds);
-    this.progress$ = this.progressBarValue(this.timeInSeconds);
-  }
-
-  private timer(value: number): Observable<number> {
-    return this.tick$.pipe(
-      withLatestFrom(this.pauseClick$),
-      filter(([_, status]) => status.active),
-      scan((acc, _) => acc - 1, value),
-      takeWhile(Boolean, true)
-    );
+    this.ts.updateTimerStatus(this.id, false);
+    this.resetStreams();
   }
 
   private progressBarValue(value: number) {
     return this.countdown$.pipe(
       map((time) => (time * 100) / value),
-      tap((progress) => {
-        if ((this.isCompleted = !progress)) {
-          this.alarm();
-        }
+      tap((time) => {
+        if (!time) this.timerProgressStatus();
       })
     );
   }
 
-  private alarm() {
-    new Audio(
-      'http://codeskulptor-demos.commondatastorage.googleapis.com/GalaxyInvaders/pause.wav'
-    ).play();
+  private setStreams() {
+    setTimeout(() => {
```

Comment:
Why so?

Reply:
Without timeout, throws Error: "expression changed after it has been checked error"

Reply:
Wonder if we can use something else to approach it.